### DISCUSSION TITLE
How does this library handle bool type variables?

### DISCUSSION THREAD

**Discussion by skleeschulte** [rlogiacco/CircularBuffer#24] [Q&A] [answered]
I wonder how this library handles bool type variables. Will each single boolean occupie one byte, or is there some optimization in place to use only one bit per boolean?

**Comment by rlogiacco** [comment, marked as answer]
The library doesn't do any particular optimization on a per data-type basis, in other words the occupied memory is `n * sizeOf(dataType)` where `n` is the circular buffer size and `sizeOf(dataType)` is the memory occupation for a particular data type. On certain architectures, a `byte` actually occupies 4 bytes and in practically any architecture I know a `bool` takes an entire byte.

If you have the need for bit manipulation you need a specialized implementation which, if done properly, is going to be a welcome addition 😉
